### PR TITLE
Suse support

### PR DESCRIPTION
**Pull request by jchengsfx**
- Installer and chef tests will use rpm in test stage for opensuse15 until the changes are merged and released.
- Installer and chef changes were manually tested on SLES 12 and 15 VMs.

### CODE REVIEW

File: deployments/chef/metadata.rb [hunk 1]
```diff
@@ -3,13 +3,15 @@
 maintainer_email 'support@signalfx.com'
 license 'Apache-2.0'
 description 'Installs/Configures the SignalFx Agent'
-version '0.2.6'
+version '0.2.7'
```
Comment (keitwb) [inline deployments/chef/metadata.rb]: I would go to 0.3.0, this is a pretty big addition.

Comment (keitwb) [reply deployments/chef/metadata.rb]: Also don't do the release increment until right before release.

File: deployments/chef/recipes/default.rb [hunk 2]
```diff
@@ -24,13 +24,34 @@
...
+  tmpdir = Dir.mktmpdir
+  package_name = if node['signalfx_agent']['package_version'].nil?
+                   'signalfx-agent'
+                 else
+                   "signalfx-agent-#{node['signalfx_agent']['package_version']}"
+                 end
+  execute 'download-agent' do
+    command "zypper --pkg-cache-dir=#{tmpdir} download #{package_name}"
+  end
+  execute 'install-agent' do
```
Comment (keitwb) [inline deployments/chef/recipes/default.rb]: This needs conditional logic so it doesn't do this every time chef runs.  Same thing with the download command above.

File: deployments/chef/metadata.rb (hunk 1 above)
Comment (jchengsfx) [reply deployments/chef/metadata.rb]: Reverted back to 0.2.6. Will increment to 0.3.0 when releasing to chef supermarket.

File: deployments/chef/recipes/default.rb (hunk 2 above)
Comment (jchengsfx) [reply deployments/chef/recipes/default.rb]: Added logic to detect installed and latest versions, and only download/install when there is a mismatch.

**Pull request merged**